From c2979a9f14add5659b0595da6181651677be76b2 Mon Sep 17 00:00:00 2001
From: highpreef <david.jorge@gmail.com>
Date: Tue, 31 Mar 2020 19:27:44 +0100
Subject: [PATCH] End-of-line compatability added

---
 src/st1920/automaton/RegExp.java | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/st1920/automaton/RegExp.java b/src/st1920/automaton/RegExp.java
index 1290c85..15f81d4 100644
--- a/src/st1920/automaton/RegExp.java
+++ b/src/st1920/automaton/RegExp.java
@@ -457,6 +457,11 @@ public class RegExp {
 		r.kind = Kind.REGEXP_ANYCHAR;
 		return r;
 	}
+	
+	private static RegExp makeEndLine() {
+		RegExp r = new RegExp("\n|\r|(\r\n)");
+		return r;
+	}
 
 	private static RegExp makeEmpty() {
 		RegExp r = new RegExp();
@@ -621,7 +626,9 @@ public class RegExp {
 	private final RegExp parseSimpleExp() throws IllegalArgumentException {
 		if (match('.'))
 			return makeAnyChar();
-		else if (check(EMPTY) && match('#'))
+		else if (match('$'))
+			return makeEndLine();
+ 		else if (check(EMPTY) && match('#'))
 			return makeEmpty();
 		else if (check(ANYSTRING) && match('@'))
 			return makeAnyString();
-- 
2.26.0.windows.1

